test/framework/src/test/java/org/elasticsearch/logsdb/datageneration/DataGeneratorSnapshotTests.java

```diff
@@ -83,22 +83,43 @@ public void testSnapshot() throws Exception {
 
         var expectedDocument = """
             {
-              "f1" : {
-                "f2" : {
-                  "f3" : "string1",
-                  "f4" : 0
+              "f1" : [
+                {
+                  "f2" : {
+                    "f3" : [
+                      null,
```

Comment:
I'm not sure if we need a mix of nulls and values like these. I think we treat nulls as if values are not provided.

Reply:
There is a difference how they are handled with synthetic source. Such null values are present in stored `_source` but not present in synthetic `_source`.

Reply:
Yes, you're right. We reserve these nulls in _source, but not in the synthetic source. Thanks for the explanation.